False positive: Pylint doesn't deduce non-None-ness from ternary condition

### Steps to reproduce

Given a file `a.py`:
```python
import random

if random.choice([0, 1]):
    x = None
else:
    x = 15

y = (-x if x is not None else None)
```

I get
```
a.py:8:5: E1130: bad operand type for unary -: NoneType (invalid-unary-operand-type)
```

### Expected behavior

It's clear that the `-x` won't be executed if `x` is None. IMO the warning is a false positive as a result.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.8.2
astroid 2.5.6
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110]
```

cc @majosm

--- a/ChangeLog
+++ b/ChangeLog
@@ -20,6 +20,9 @@ What's New in Pylint 2.13.7?
 ============================
 Release date: TBA
 
+* Fix a crash caused by using the new config from 2.14.0 in 2.13.x code.
+
+  Closes #6408
 
 
 What's New in Pylint 2.13.6?

--- a/doc/release.md
+++ b/doc/release.md
@@ -71,7 +71,8 @@ cherry-picked on the maintenance branch.
 - Bump the version and release by using `tbump X.Y-1.Z --no-push`. (For example:
   `tbump 2.3.5 --no-push`)
 - Check the result visually with `git show`.
-- Push the tag.
+- Open a merge request to run the CI tests for this branch
+- Create and push the tag.
 - Release the version on GitHub with the same name as the tag and copy and paste the
   appropriate changelog in the description. This triggers the PyPI release.
 - Merge the `maintenance/X.Y.x` branch on the main branch. The main branch should have

--- a/pylint/checkers/variables.py
+++ b/pylint/checkers/variables.py
@@ -2069,9 +2069,7 @@ class VariablesChecker(BaseChecker):
         if not isinstance(defstmt, nodes.AnnAssign) or defstmt.value:
             return False
 
-        if node.name in self.linter.config.additional_builtins or utils.is_builtin(
-            node.name
-        ):
+        if node.name in self.config.additional_builtins or utils.is_builtin(node.name):
             return False
 
         defstmt_frame = defstmt.frame(future=True)

